cmd/docker/docker.go

```diff
@@ -325,6 +326,25 @@ func tryPluginRun(ctx context.Context, dockerCli command.Cli, cmd *cobra.Command
 	return nil
 }
 
+// registerForceExitGoroutine registers a goroutine that will force exit the
+// process after 3 SIGTERM/SIGINT signals.
+func registerForceExitGoroutine(ctx context.Context, w io.Writer) {
+	// setup a signal handler to force exit after 3 SIGTERM/SIGINT
+	go func() {
+		<-ctx.Done()
```

Comment:
Since this is expected to be a fallback mechanism - I think we can just rely on catching the signals directly instead of relying on this to receive the first cancellation.

Reply:
Either this, or we could collapse the now two places where we're receiving signals (the main mechanism, set up in https://github.com/laurazard/cli/blob/bc291f0c987156364a7db2ca79f791e5cf387174/cmd/docker/docker.go#L38) and this one, and instead handle everything there? This seems simpler than registering two separate signal handlers, and condenses the signal handling logic into one place, which I prefer.

Something like:
```go
ctx, cancel := context.WithCancel(context.Background())
	sigChan := make(chan os.Signal, 3)
	signal.Notify(sigChan, platformsignals.TerminationSignals...)
	go func() {
		cancelCounter := 0
		for range sigChan {
			if cancelCounter >= 2 {
				os.Exit(1)
			}
			cancel()
			cancelCounter++
		}
	}()
```

Reply:
This is a cleaner solution, however, what about plugins though? Since we have our own signal handler for plugins and the strategy of killing plugins is separate from the main CLI commands.

Reply:
Ah, right 🤦‍♀️ nevermind my comment then, I guess at least for now we should stay consistent with the plugin pattern and register a new signal handler after the main context is cancelled.